lib/Dialect/MSFT/MSFTPasses.cpp

```diff
@@ -756,25 +759,32 @@ void PartitionPass::bubbleUpGlobalRefs(
       return;
     assert(oldPath.size() > 1);
 
-    // Construct a new path starting from the old path.
-    SmallVector<Attribute> newPath(oldPath.begin(), oldPath.end());
-
-    // Use the elements in the path to construct a name that matches the format
-    // going into inner_ref above. Splice the last two elements in the path,
-    // bulding the name from the two elements' names, and taking the module from
-    // the second to last element. Note that we use the global ref directly so
-    // we can update all paths to an instance the first time it is visited. This
-    // saves on bookkeeping to match up the new inner_ref to the global refs.
-    // TODO: consider adding state to simplify this.
-    auto lastElement = newPath.pop_back_val().cast<hw::InnerRefAttr>();
-    auto nextElement = newPath.pop_back_val().cast<hw::InnerRefAttr>();
-    auto newModule = nextElement.getModule();
-    auto lastName = lastElement.getName().getValue();
-    auto nextName = nextElement.getName().getValue();
-    auto newNameAttr =
-        StringAttr::get(op->getContext(), nextName + "." + lastName);
-    auto newLeaf = hw::InnerRefAttr::get(newModule, newNameAttr);
-    newPath.push_back(newLeaf);
+    // Find the index of the node in the path that points to the opName. The
+    // previous node in the path must point to parentName.
+    size_t opIndex = 0;
+    for (; opIndex < oldPath.size(); ++opIndex)
+      if (oldPath[opIndex].cast<hw::InnerRefAttr>().getName() == oldInnerSym)
```

Comment:
What if there is another instance in the path list which happens to have the same instance symbol? Don't you need to match on both the module and inner_sym?